test/lit/source-map.wast

```diff
@@ -12,8 +15,11 @@
         ;;@ src.cpp:40:1
         (local.get $y)
       )
+      ;; For the legacy parser
       ;;@ src.cpp:50:1
       (then
+        ;; For the new parser
+        ;;@ src.cpp:50:1
         (return)
```

Comment:
Looks like the old parser doesn't roundtrip this properly and the new one does, is that correct? Perhaps add a comment to clarify that so it's obvious why we have these two 50:1 annotations?

Reply:
They can both round-trip properly, but the old one only if the annotation is on the `then` and the new one only if the annotation is on the `return`. Do you think this kind of breaking change is going to be an issue? We can make sure j2wasm is doing the right thing, but other users would be more difficult.

Reply:
Is it hard to match the old behavior?

This doesn't seem like a dangerous breaking change to me, as our text format in general is not entirely standard nor stable, but we should document it.

Reply:
Sounds ok to me.

Reply:
Do you have a suggestion about where to document it? The changelog might make sense, but more so if we were actually enabling the new parser at the same time.

Reply:
Yes, the changelog. I agree it makes more sense when we enable the new parser. Maybe we can put a note in the changelog up above the current release notes, in preparation for enabling the new parser, to remind us, or something like that.

Reply:
Done.